pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -449,6 +455,16 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyNetworkView map[string]b
 	return localityLbEndpoints
 }
 
+// addUint32AvoidOverflow returns sum of two uint32 and status. If sum overflows,
+// and returns MaxUint32 and status.
+func addUint32(left, right uint32) (uint32, bool) {
+	newVal := left + right
+	if newVal < left || newVal < right {
```

Comment:
Sorry for the late reply .... Could this just be simplified to:

```golang
if math.MaxUint32 - right < left {
    return math.MaxUint32, true
}
return left + right, false
```

Reply:
yes, but this PR is merged. I will bring this improvement to the next PR.

Reply:
Sounds good ... thanks for following up.